controllers/applicationvolumereplication_controller.go

```diff
@@ -269,6 +432,11 @@ func (r *ApplicationVolumeReplicationReconciler) extractHomeClusterAndPeerCluste
 		return "", "", fmt.Errorf("failed to get cluster map for placement %s error: %w", placementRule.Name, err)
 	}
 
+	err = r.filterClusters(placementRule, clmap)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to filter clusters. Cluster len %d, error (%w)", len(clmap), err)
+	}
+
 	if len(clmap) != clusterCount {
```

Comment:
(more around line 451 below) Should we add another check here to enforce/error-out when ClusterReplicas is not 1

Reply:
We definitely should